src/unix/blockq.c

```diff
@@ -227,6 +228,7 @@ sysreturn blockq_check_timeout(blockq bq, thread t, blockq_action a,
 
     bi->a = a;
     bi->t = t;
+    refcount_reserve(&t->refcount);
```

Comment:
Should probably be released if the call to register_timer() fails a couple of lines below

Reply:
Thanks; I added a check for register_timer failure.